### PR TITLE
feat(web): selectAllLabel with showCount

### PR DESCRIPTION
**Pull request by SavvyShah**
`selectAllLabel` label should show count as total number of documents when `showCount` is true.

- [Loom preview](https://www.loom.com/share/0a83a7e15fa54473995d7dc60a256904)
- [Notion card](https://www.notion.so/appbase/selectAllLabel-support-for-Elasticsearch-0afc7e0e88f84f998dbee01b6839bc0a)

- [Reactivecore PR](https://github.com/appbaseio/reactivecore)

### CODE REVIEW

File: packages/web/src/components/list/SingleDataList.js [hunk 1]
```diff
@@ -398,15 +402,27 @@ class SingleDataList extends Component {
 									id={`${this.props.componentId}-${selectAllLabel}`}
 									value={selectAllLabel}
 									tabIndex={isAllChecked ? '-1' : '0'}
-									onChange={this.handleClick}
+									onClick={this.handleClick}
```
Comment (bietkul) [inline packages/web/src/components/list/SingleDataList.js]: @SavvyShah What is the reason of this change?

File: packages/web/src/components/list/TabDataList.js [hunk 2]
```diff
@@ -36,7 +39,11 @@ const TabDataList = (props) => {
 							vertical={props.displayAsVertical}
 							key={props.selectAllLabel}
 						>
-							{props.selectAllLabel}
+							{defaultItem({
+								label: props.selectAllLabel,
+								value: props.selectAllLabel,
+								count: props.showCount && getNestedProperty(rawData, 'hits.total.value'),
```
Comment (bietkul) [inline packages/web/src/components/list/TabDataList.js]: @SavvyShah Fix it, use `total` from redux

File: packages/web/src/utils/index.js [hunk 3]
```diff
@@ -282,3 +282,16 @@ export function decodeHtml(str) {
 		return String.fromCharCode(num);
 	});
 }
+
+// Similar to lodash.get eg. a[0].b.c
+export function getNestedProperty(object, keys, defaultVal = null): any {
```
Comment (bietkul) [inline packages/web/src/utils/index.js]: @SavvyShah I think it is not required

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/web/src/components/list/SingleDataList.js (hunk 1 above)
Comment (SavvyShah) [reply packages/web/src/components/list/SingleDataList.js]: Because it was used by other [list items rendered through `data` property](https://github.com/appbaseio/reactivesearch/blob/33fa468f3c15f1f61431373cf7c53312a28d7304/packages/web/src/components/list/SingleDataList.js#L446), so to create consistency.

File: packages/web/src/components/list/TabDataList.js [hunk 4]
```diff
@@ -27,7 +28,9 @@ const TabDataList = (props) => {
 		<SingleDataList
 			{...props}
 			showSearch={props.showSearch}
-			render={({ data, value, handleChange }) => (
+			render={({
```
Comment (bietkul) [inline packages/web/src/components/list/TabDataList.js:31]: @SavvyShah You don't need to connect this component, provide `total` value in the render method of `SingleDataList` and use it.

**Review by bietkul — CHANGES_REQUESTED** [review]

**Review by bietkul — APPROVED** [review]
